build/generate/featureMatrix.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"bytes"
+	"html/template"
+	"io/ioutil"
+	"sort"
+
+	"github.com/StackExchange/dnscontrol/providers"
+	_ "github.com/StackExchange/dnscontrol/providers/_all"
+)
+
+func generateFeatureMatrix() error {
+	allNames := map[string]bool{}
+	for n := range providers.RegistrarTypes {
+		allNames[n] = true
+	}
+	for n := range providers.DNSProviderTypes {
+		allNames[n] = true
+	}
+	providerTypes := []string{}
+	for n := range allNames {
+		providerTypes = append(providerTypes, n)
+	}
+	sort.Strings(providerTypes)
+	matrix := &FeatureMatrix{
+		Providers: map[string]FeatureMap{},
+		Features: []FeatureDef{
+			{"Official Support", "This means the provider is actively used at stack exchange, and we offer a stronger guarantee it will work"},
```

Comment:
actively used at Stack Exchange, bugs are more likely to be fixed, and failing integration tests will block a release. See docs for details.